src/devices/machine/t10mmc.cpp

```diff
@@ -360,12 +366,16 @@ void t10mmc::ExecCommand()
 			if (end_track > command[7])
 				end_track = command[7];
 
-			// HACK: assume index 0 & 1 means beginning of track and anything else means end of track
-			if (command[8] <= 1)
-				end_track--;
+			// konamigv lacrazyc just sends same track start/end
+			if (command[4] != command[7] && command[5] != command[8])
+			{
+				// HACK: assume index 0 & 1 means beginning of track and anything else means end of track
+				if (command[8] <= 1)
+					end_track--;
```

Comment:
Note: of course the actual fix would be to actually implement a `cdrom_device->get_track_end` at very least.

Reply:
It should scan forward until it finds the requested index in the subchannel data or encounters the end of the track.

Reply:
Making cdrom_device to comply with seek / scan states is interesting in itself, but barely testable from CD-ROM player BIOSes (where the only concrete test case afaik is Zero Divide on Saturn).

Reply:
Anyway, my main point is that it isn’t aware of track end as such.  The reason index 2 (or higher) stops at the end of the track in most cases is because the track only contains index 1, so it stops when it encounters the first sector in the next track (or the end of the session if it’s the last track in the session).  If the track actually contains index 2 or higher, it will stop on the first sector containing the index specified in the command.

Reply:
I'm aware, what I'm saying here is that subchannel stuff itself is still in the "parent does something but clients do whatever they want with them" phase, something that eventually needs to be done (particularly for CD+G support) but I currently don't have the willpower to look at Neo Geo CD / Mega CD one first.